File: 백준/Silver/10845. 큐/큐.py
import sys
line = int(input())
queue=[]

for i in range(line):
    alist=sys.stdin.readline().split()
    order=alist[0]
    
    if order=="push":
        queue.append(alist[1])
        
    elif order=="pop":
        if len(queue)>0:
            print(queue[0])
            del queue[0]
        else:
            print("-1")
        
    elif order=="size":
        print(len(queue))
    elif order=="empty":
        if len(queue)>0:
            print("0")
        else:
            print("1")
    elif order=="front":
        if len(queue)>0:
            print(queue[0])
        else:
            print("-1")
    elif order=="back":
        if len(queue)>0:
            print(queue[-1])
        else:
            print("-1")
            

File: 백준/Silver/10845. 큐/큐.java

    import java.io.IOException;
    import java.io.BufferedReader;
    import java.io.InputStreamReader;
    import java.util.LinkedList;


    public class Main {
        public static void main(String[] args) throws IOException {

            BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
            int len = Integer.parseInt(reader.readLine());
            queue que= new queue();
            for (int i = 0; i < len; i++) {
                String command = reader.readLine();
                if(command.startsWith("push")){
                    int value=Integer.parseInt(command.split(" ")[1]);
                    que.push(value);
                }else if(command.equals("pop")){
                    System.out.println(que.pop());
                }else if(command.equals("size")){
                    System.out.println(que.size());
                }else if (command.equals("empty")){
                    if(que.empty()){
                    System.out.println("1");
                    }else{
                        System.out.println("0");
                    }
                } else if (command.equals("front")) {
                    System.out.println(que.front());
                } else if (command.equals("back")) {
                    System.out.println(que.back());
                }


            }
        }

    }
    class queue{
        LinkedList<Integer> que =new LinkedList<>();


        public void push(int value){
            que.add(value);
        }
        public int pop(){
            if(que.isEmpty()){
                return -1;
            }
           return que.pop();
        }
        public int size(){
            return que.size();
        }
        public boolean empty(){
            return que.isEmpty();
        }
        public int front(){
            if(que.isEmpty()){
                return -1;
            }
            return que.getFirst();
        }
        public int back(){
            if(que.isEmpty()){
                return -1;
            }
            return que.getLast();
        }
    }